src/controllers/controller.h

```diff
@@ -193,13 +196,33 @@ class ControllerJSProxy : public QObject {
             : m_pController(m_pController) {
     }
 
+    QHash<QString, QString> m_cache;
+
     // The length parameter is here for backwards compatibility for when scripts
     // were required to specify it.
     Q_INVOKABLE virtual void send(const QList<int>& data, unsigned int length = 0) {
         Q_UNUSED(length);
         m_pController->send(data, data.length());
     }
 
+    // Available charsets should be available here:
+    // http://www.iana.org/assignments/character-sets/character-sets.xhtml
+    Q_INVOKABLE QByteArray convertEncoding(const QString& targetCharset, const QString& value) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+        auto* codec = QTextCodec::codecForName(targetCharset.toUtf8());
+        if (!codec) {
+            return nullptr;
+        }
+        return codec->fromUnicode(value);
+#else
+        QStringEncoder fromUtf8 = QStringEncoder(targetCharset.toUtf8().data());
```

Comment:
`QStringEncoder` and `QTextCodec` seem to allocate resources that can be reused. Is it a good idea, performance-wise, to cache them in a `QMap` or something?

Reply:
I don't think so. From what I can tell (by only looking at the docs, not the source), it supports encoding in chunks instead of all at once, so it needs internal state there. That state should probably not be reused across unrelated strings.

Reply:
There's a `resetState` method that can be called first, though.

Reply:
yeah, but I don't think its worth the complexity. Taking a quick glance at the source, there is no heap allocation or anything else expensive that is worth reusing. We can KISS for now and optimize later.

Reply:
Ok. this conversation is resolved then.